cmd/kubeadm/app/phases/upgrade/compute_test.go

```diff
@@ -995,3 +1099,66 @@ func TestGetSuggestedEtcdVersion(t *testing.T) {
 		})
 	}
 }
+
+func newMockClientForTest(t *testing.T, dnsName string, dnsVersion string, multiDNS bool) *clientsetfake.Clientset {
+	client := clientsetfake.NewSimpleClientset()
+	_, err := client.AppsV1().Deployments(metav1.NamespaceSystem).Create(context.TODO(), &apps.Deployment{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Deployment",
+			APIVersion: "apps/v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      dnsName,
+			Namespace: "kube-system",
+			Labels: map[string]string{
+				"k8s-app": "kube-dns",
+			},
+		},
+		Spec: apps.DeploymentSpec{
+			Template: v1.PodTemplateSpec{
+				Spec: v1.PodSpec{
+					Containers: []v1.Container{
+						{
+							Image: "test:" + dnsVersion,
+						},
+					},
+				},
+			},
+		},
+	}, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatalf("error creating deployment: %v", err)
+	}
+
+	if multiDNS {
+		_, err := client.AppsV1().Deployments(metav1.NamespaceSystem).Create(context.TODO(), &apps.Deployment{
```

Comment:
you can just create a local function inside newMockClientForTest for example

```
createDeployment := func(....some args) {
}
```

and use it two times:

err := createDeployment(name1....some other args)
...
err := createDeployment(name2....some other args)

this will reduce the lines of code by 2x